crates/prune/src/builder.rs

```diff
@@ -51,6 +58,15 @@ impl PrunerBuilder {
         self
     }
 
+    /// Sets the timeout for pruner, per run.
+    ///
+    /// CAUTION: Account and Storage History prune segments treat this timeout as a soft limit,
```

Comment:
Why?

Reply:
we do account/storage history pruning in two steps:
1. Prune the changesets and memorize the highest pruned block
2. Prune indices up to the highest pruned block from changesets

Given that, we can't stop the step 2 if time limit is reached, because then we'll have inconsistent data in the database (partly pruned block).

Reply:
It needs to be solved separately because not obvious yet how to do that and the PR is already large